airbyte-integrations/connectors/source-slack/source_slack/manifest.yaml

```diff
@@ -253,11 +253,75 @@ definitions:
               - channel_id
             value: "{{ stream_partition.get('channel') }}"
 
+  threads_stream:
+    $ref: "#/definitions/stream_base"
+    $parameters:
+      name: threads
+      path: conversations.replies
+      extractor_field_path: messages
+    primary_key:
+      - channel_id
+      - ts
+    retriever:
+      $ref: "#/definitions/retriever"
+      requester:
+        $ref: "#/definitions/requester"
+        request_parameters:
+          channel: "{{ stream_slice['parent_slice']['channel'] }}"
+        error_handler:
+          type: DefaultErrorHandler
+          backoff_strategies:
+            - type: "WaitTimeFromHeader"
+              header: "retry-after"
+            - type: "WaitTimeFromHeader"
+              header: "Retry-After"
+      record_selector:
+        $ref: "#/definitions/selector"
+      paginator:
+        $ref: "#/definitions/default_paginator"
+      partition_router:
+        type: SubstreamPartitionRouter
+        parent_stream_configs:
+          - type: ParentStreamConfig
+            stream:
+              $ref: "#/definitions/channel_messages_stream"
+            parent_key: ts
+            partition_field: float_ts
+            request_option:
+              type: RequestOption
+              field_name: "ts"
+              inject_into: "request_parameter"
+    incremental_sync:
+      type: DatetimeBasedCursor
```

Comment:
Do we need a state transformation from the incoming legacy state to the per-partition state since this is now a concurrent incremental + partition router stream?

Reply:
substream partition router has built in logic to migrate state [here](https://github.com/airbytehq/airbyte-python-cdk/blob/main/airbyte_cdk/sources/declarative/partition_routers/substream_partition_router.py#L281)